server/src/server.ts

```diff
@@ -26,9 +26,39 @@ import { assert } from "console";
 import { WorkspaceEdit } from "vscode-languageserver";
 import { onErrorReported } from "./errorReporter";
 import * as ic from "./incrementalCompilation";
-import config, { extensionConfiguration } from "./config";
+import config, { extensionConfiguration, initialConfiguration } from "./config";
 import { projectsFiles } from "./projectFiles";
 import { NormalizedPath } from "./utils";
+import { initializeLogger, getLogger, setLogLevel, LogLevel } from "./logger";
+
+function applyUserConfiguration(configuration: extensionConfiguration) {
+  // We always want to spread the initial configuration to ensure all defaults are respected.
+  config.extensionConfiguration = Object.assign(
+    {},
+    initialConfiguration,
+    configuration,
+  );
+
+  const debugLoggingEnabled =
+    configuration.incrementalTypechecking?.debugLogging === true;
+
+  if (debugLoggingEnabled) {
+    // incrementalTypechecking.debugLogging overrides logLevel to retain legacy behavior
+    setLogLevel("log");
+    return;
+  }
```

Comment:
I don't think we need to retain legacy behavior here. I can't imagine anyone else but us using that property anyway.